### PR TITLE
Upgrade web3 version to 1.2.4

### PR DESCRIPTION
**Pull request by manhlx3006**

### CODE REVIEW

File: test/conversionRate.js [hunk 1]
```diff
@@ -3,10 +3,15 @@ const TestToken = artifacts.require("TestToken.sol");
 const Wrapper = artifacts.require("./mockContracts/Wrapper.sol");
 
 const Helper = require("./helper.js");
-const BigNumber = require('bignumber.js');
+const BN = web3.utils.BN;
+
+require("chai")
+    .use(require("chai-as-promised"))
```
Comment (ilanDoron) [inline test/conversionRate.js]: I thought chai is included in new web3. no?

File: test/conversionRate.js [hunk 2]
```diff
@@ -79,8 +84,7 @@ contract('ConversionRates', function(accounts) {
 
         for (let i = 0; i < 14; i++) {
             byte = await wrapper.getInt8FromByte(hexArr, i);
-//            console.log("byte " + i + ": " + byte.valueOf());
-            assert.equal(byte.valueOf(), arr[i].valueOf(), "bad bytes 14. index: " + i);
+            Helper.assertEqual(byte, arr[i], "bad bytes 14. index: " + i);
```
Comment (ilanDoron) [inline test/conversionRate.js]: nice
Seems like not that hard to do this one.

File: package.json [hunk 3]
```diff
@@ -37,13 +37,14 @@
...
-    "web3": "^0.20.7",
-    "web3-utils": "^1.2.1",
+    "web3": "^1.2.4",
+    "web3-utils": "^1.2.4",
     "yargs": "^11.0.0"
   },
   "devDependencies": {
     "@nomiclabs/buidler": "^1.0.1",
     "chai": "^4.1.2",
+    "chai-bn": "^0.2.0",
```
Comment (ilanDoron) [inline package.json]: do we need chai-bn in separate package
though it was part of the environment...

File: test/v4/conversionRate.js [hunk 4]
```diff
@@ -361,18 +360,18 @@ contract('ConversionRates', function(accounts) {
 
         // get rate without activating quantity step function (small amount).
         let srcQty = 2;
-        let expectedRate = (new BigNumber(baseBuyRate));
+        let expectedRate = (new BN(baseBuyRate));
         let extraBps = compactBuyArr1[tokenInd] * 10;
-        expectedRate = addBps(expectedRate, extraBps);
-        let dstQty = new BigNumber(srcQty).mul(expectedRate).div(precisionUnits);
+        expectedRate = Helper.addBps(expectedRate, extraBps);
+        let dstQty = new BN(srcQty).mul(expectedRate).div(precisionUnits);
```
Comment (ilanDoron) [inline test/v4/conversionRate.js]: shouldn't we use dstQty function here?
for edge cases?
